### PR TITLE
feat: find enrichments by value leveraging p_match field

### PR DESCRIPTION
**Pull request by edyesed**
### Background

p_match has rolled out as part of enrichment, and that allows us to traverse enrichments by value.

### Changes

extends the base class for lookup tables to walk all lookup tables on the event, collecting matches based on p_match value

### Testing

added global helper unit tests

### CODE REVIEW

**Comment by edyesed** [comment]
@rleighton / @debugmiller -> let me know what you think about this approach, happy to make changes. This is a bit unlike the historical approach to inspecting enrichment(s).

File: global_helpers/panther_lookuptable_helpers.py [hunk 1]
```diff
@@ -14,3 +15,16 @@ def _lookup(self, match_field: str, *keys) -> list or str:
         if isinstance(match, list):
             return [deep_get(match_value, *keys) if match_value else None for match_value in match]
         return deep_get(match, *keys)
+
+    def enrichments_by_pmatch(self, event, p_match: str) -> dict:
```
Comment (rleighton) [inline global_helpers/panther_lookuptable_helpers.py:19]: Can you add a comment string explaining what this does and what is returned.

File: global_helpers/panther_lookuptable_helpers.py [hunk 2]
```diff
@@ -14,3 +15,16 @@ def _lookup(self, match_field: str, *keys) -> list or str:
...
+        matched_items = {}
+        for enrichment_type in deep_get(event, ENRICHMENT_KEY, default={}).keys():
```
Comment (rleighton) [inline global_helpers/panther_lookuptable_helpers.py:21]:
```suggestion
        for lut_name in deep_get(event, ENRICHMENT_KEY, default={}).keys():
```

File: global_helpers/global_helpers_test.py [hunk 3]
```diff
@@ -1772,5 +1793,69 @@ def test_alert_context(self):
...
 
+class TestLookupTableHelpers(unittest.TestCase):
+    # pylint: disable=protected-access
+    def setUp(self):
+        self.simple_event_no_pmatch = {
+            "p_enrichment": {"tor_exit_nodes": {"foo": {"ip": "1.2.3.4"}}}
+        }
+        self.simple_event = {
+            "p_enrichment": {
+                "tor_exit_nodes": {"foo": {"ip": "1.2.3.4", "p_match": "1.2.3.4"}},
```
Comment (rleighton) [inline global_helpers/global_helpers_test.py:1804]: Ensure that this handles more than 1 key
```suggestion
                "tor_exit_nodes": {
                   "foo": {"ip": "1.2.3.4", "p_match": "1.2.3.4"},
                   "bar": {"ip": "1.2.3.5", "p_match": "1.2.3.5"}
                  },
```

File: global_helpers/panther_lookuptable_helpers.py (hunk 1 above)
Comment (rleighton) [inline global_helpers/panther_lookuptable_helpers.py:19]: `enrichments_by_pmatch` is descriptive but long. Is there a shorter name that is still conveys the action?

File: global_helpers/panther_lookuptable_helpers.py [hunk 4]
```diff
@@ -14,3 +18,32 @@ def _lookup(self, match_field: str, *keys) -> list or str:
         if isinstance(match, list):
             return [deep_get(match_value, *keys) if match_value else None for match_value in match]
         return deep_get(match, *keys)
+
+    @property
+    def p_matched(self):
```
Comment (rleighton) [inline global_helpers/panther_lookuptable_helpers.py:23]: nit: Unit test?

**Review by rleighton — APPROVED** [review]

File: global_helpers/panther_lookuptable_helpers.py (hunk 1 above)
Comment (edyesed) [reply global_helpers/panther_lookuptable_helpers.py]: yep, now called `p_matches`

File: global_helpers/panther_lookuptable_helpers.py (hunk 4 above)
Comment (edyesed) [reply global_helpers/panther_lookuptable_helpers.py:23]: there are a few of them, [here's one](https://github.com/panther-labs/panther-analysis/blob/7eec5df10c687164809d8a1e561c32dd9bf168ad/global_helpers/global_helpers_test.py#L1871)

**Review by andrea-youwakim — APPROVED** [review]

**Pull request merged**